### ISSUE TITLE
laravel: unable to get session information due to error Missing Authorization key in headers array

### ISSUE THREAD

**Issue by bilal7446** [Shopify/shopify-cli#1950]
# Issue summary
i created laravel project using shopify cli on localhost. when app installation is complete i tried running url /rest-example to run example of rest api call but i got into issue where script was unable to get Authorization key in header array. i searched everywhere for issue but most solution require you to write some code in .htaccess but that doesnt work. 

## Expected behavior
Get shopify session object

What do you think should happen?
somehow server is preventing authorization key to be sent in header array and i dont know how to make it send authorization key

## Actual behavior
error received 

Tip: include an error message (in a `<details></details>` tag) if your issue is related to an error

Shopify\Exception\MissingArgumentException
Missing Authorization key in headers array

## Specifications

- App type: Laravel
- Operating System: windows 10
- Ruby version (ruby -v): 2.7

**Comment by fl4r** [comment]
Hello,

I have the same error, I can't use the Rest Api in the Laravel Shopify Public App (form the cli). For the /rest-example route, I got the following error :
`Shopify\Exception\MissingArgumentException
Missing Authorization key in headers array`

So i tried to do the same like in the Route::get('/rest-example' ...) , for the Route::fallback (in routes/web.php) :

`Route::fallback(function (Request $request) {
    /** @var AuthSession */
    $session = $request->get('shopifySession'); // Provided by the shopify.auth middleware, guaranteed to be active

    $client = new Rest($session->getShop(), $session->getAccessToken());
    $result = $client->get('products', [], ['limit' => 5]);

    return response($result->getDecodedBody());
})->middleware('shopify.auth:online');`

But the same first error happened :
`Shopify\Exception\MissingArgumentException
Missing Authorization key in headers array`

Thx, I'd like to know more about it


(App type: Laravel)
(Operating System: mac OS Big Sur)
(Homebrew 3.3.12)
(PHP 7.4.27)

**Comment by cennis-ck** [comment]
> Hello,
> 
> I have the same error, I can't use the Rest Api in the Laravel Shopify Public App (form the cli). For the /rest-example route, I got the following error : `Shopify\Exception\MissingArgumentException Missing Authorization key in headers array`
> 
> 
> So i tried to do the same like in the Route::get('/rest-example' ...) , for the Route::fallback (in routes/web.php) :
> 
> `Route::fallback(function (Request $request) { /** @var AuthSession */ $session = $request->get('shopifySession'); $client = new Rest($session->getShop(), $session->getAccessToken()); $result = $client->get('products', [], ['limit' => 5]); return response($result->getDecodedBody()); })->middleware('shopify.auth:online');`
> 
> But the same first error happened : `Shopify\Exception\MissingArgumentException Missing Authorization key in headers array`
> 
> 
> Thx, I'd like to know more about it
> * (App type: Laravel)
> * (Operating System: mac OS Big Sur)
> * (Homebrew 3.3.12)
> * (PHP 7.4.27)

did you solve this? i'm so sick of fighting all these shopify client apis with their vague information. it should not be this hard

**Comment by brightraudona** [comment]
> Hi,
> 
> You need to add the authorization token for Axios request, so get the token in the frontend site using AppBrige like this:
> 
> ```js
>    const app = useAppBridge();                                                                                                                                                  
>    getSessionToken(app).then(async (token) => {                                                                                                                                          
>            const response = await axios.post('/xxx', {}, {                                                                                                                           
>                headers: {                                                                                                                                                                    
>               Authorization: 'Bearer ' + token                                                                                                                                              
>              }                                                                                                                                                                               
>            });                                                                                                                                                                               
>     }); 
> ```
> 
> Also remember to avoid the Laravel csrf token verification for that route in the file VerifyCsrfToken.php.
> 
> I hope that can be useful for you

this is just too vague, where should this be added? @eddymdz

**Comment by cardfaux** [comment]
Here is how I implemented the above information.

1. I made a route in the /routes/api.php
`Route::post('/createlink', 'App\Http\Controllers\LinkController@store')->middleware('shopify.auth:online');`
2. In the CreateNewProductLink.jsx file where the post request is made with axios, I put.

`
import {useAppBridge,} from "@shopify/app-bridge-react";
import { getSessionToken } from "@shopify/app-bridge-utils";
import axios from "axios";

const appBridge = useAppBridge();

the clickedSaveButton function is the function that gets triggered when a button is clicked.
async function clickedSaveButton() {
  const sessionToken = await getSessionToken(appBridge);

  axios.post("/api/createlink", {
    {data_to_send: "the data that needs sent", more_data_to_send: "more data to send"},
    {headers: {Authorization: "Bearer " + `${sessionToken}`,}}
  }).then(function (response) {
      console.log('the data was sent', response);
    }).catch(function (error) {
        console.log('made it into the error block', error);
      });
}
`
